ts/private/ts_project_worker.js

```diff
@@ -265,28 +426,35 @@ function createFilesystemTree(root, inputs) {
         const parts = p.split(path.sep);
         let node = watchingTree;
         for (const part of parts) {
+            if (!part) {
+                continue;
+            }
             if (!(part in node)) {
                 node[part] = {};
             }
             node = node[part];
         }
         if (!(Watcher in node)) {
             node[Watcher] = new Set();
-        }  
-        const watcher = {callback, recursive};
+        }
+        const watcher = { callback, recursive };
         node[Watcher].add(watcher);
         return () => node[Watcher].delete(watcher)
     }
 
-    return { add, remove, update, fileExists, directoryExists, readDirectory, getDirectories, watchDirectory: watch, watchFile: watch }
+    function watchFile(p, callback) {
+        return watch(p, callback)
+    }
+
+    return { add, remove, update, notify, fileExists, directoryExists, isSymlink, realpath, readDirectory, getDirectories, watchDirectory: watch, watchFile: watchFile, printTree }
 }
 
 
 /** Program and Caching */
 function isExternalLib(path) {
-    return  path.includes('external') && 
-            path.includes('typescript@') && 
-            path.includes('node_modules/typescript/lib')
+    return path.includes('external') &&
```

Comment:
what is an external lib in this context?

Reply:
is a file that is transitively depended on and comes from an external bazel workspace which `npm_typescript` in this case.